packages/kit/test/apps/basics/src/routes/load/invalidation/invalidate-then-goto/+page.svelte

```diff
@@ -0,0 +1,23 @@
+<script>
+	import { invalidate, goto } from '$app/navigation';
+
+	/** @type {import('./$types').PageData} */
+	export let data;
+</script>
+
+<p class="layout">{data.layoutDate}</p>
+<p class="page">{data.pageDate}</p>
+<button
+	type="button"
+	class="invalidate"
+	on:click={() => (window.promise = invalidate('invalidate-depends:goto'))}
```

Comment:
why isn't this just?
```suggestion
	on:click={async () => await invalidate('invalidate-depends:goto')}
```

if it needs to be two separate statements the test could use a comment explaining

making this `window.promise` makes me a bit nervous you might inadvertently end up with two tests using that name that stomp on each other. I feel like you might almost need to make it be something like `window.invalidate_then_goto_promise`

Reply:
See the corresponding test - this global is awaited. You're right that it should be a unique name.

Reply:
Right, I saw that it's awaited. I guess my question is will Svelte not await async handlers? There's no mention on https://svelte.dev/docs/element-directives of what you should do if you need to call an async function from a click handler. This is something I actually encountered just yesterday in real Svelte code and felt a bit lost as to what would/should happen there. We might need to update those docs a bit. Is this how users should write it in their code as well or is some of the way we're writing it here caused by invoking it through Playwright?

Reply:
But this isn't what this is about at all, your suggestion would break the test because the test awaits `window.promise`, and your change would make so that this is no longer the case.